### PR TITLE
[dev-client] Use the new autolinking instead of the one from RN

### PR DESCRIPTION
**Pull request by lukmccall**
# Why

Closes ENG-2317
Closes ENG-1695

# How

- Moved `podspec` to `ios` directory (required by our auto-linking).
- Restructured `expo-dev-menu` codebase on ios to work with a new location of `podspec`.
- `dependecies.js` is exporting an empty object.
-  Added `expo` as a dependency of `dev` packages.
- Changed how modules are exported on Android (This will be refactored in the future, but for now this should be good enough). 

# Test Plan

- bare-expo ✅
- fresh app with `expo-updates installed ✅

### CODE REVIEW

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/DevLauncherPackage.kt [hunk 1]
```diff
@@ -7,8 +7,9 @@ import com.facebook.react.uimanager.ViewManager
 import expo.modules.devlauncher.modules.DevLauncherDevMenuExtensions
 import expo.modules.devlauncher.modules.DevLauncherInternalModule
 import expo.modules.devlauncher.modules.DevLauncherModule
+import expo.modules.core.interfaces.Package
 
-class DevLauncherPackage : ReactPackage {
+class DevLauncherPackage : Package, ReactPackage {
```
Comment (tsapeta) [inline packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/DevLauncherPackage.kt:12]: Is `ReactPackage` necessary here?
I think we use `BasePackage` elsewhere 😉

File: packages/expo-dev-menu-interface/expo-module.config.json [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{
```
Comment (tsapeta) [inline packages/expo-dev-menu-interface/expo-module.config.json:1]: Do you plan on moving `expo-dev-menu-interface` to the core? Would we benefit of it?

File: packages/expo-dev-menu-interface/package.json [hunk 3]
```diff
@@ -21,5 +21,8 @@
   },
   "author": "650 Industries, Inc.",
   "license": "MIT",
-  "homepage": "https://docs.expo.io"
+  "homepage": "https://docs.expo.io",
```
Comment (tsapeta) [inline packages/expo-dev-menu-interface/package.json:24]:
```suggestion
  "homepage": "https://docs.expo.dev",
```

File: packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/DevMenuPackage.kt [hunk 4]
```diff
@@ -10,16 +11,20 @@ import expo.modules.devmenu.modules.DevMenuInternalModule
 import expo.modules.devmenu.modules.DevMenuManagerProvider
 import expo.modules.devmenu.modules.DevMenuModule
 import expo.modules.devmenu.modules.DevMenuSettings
+import expo.modules.core.interfaces.Package
 
-class DevMenuPackage : ReactPackage {
-  override fun createNativeModules(reactContext: ReactApplicationContext) =
-    listOf(
+class DevMenuPackage: Package, ReactPackage {
```
Comment (tsapeta) [inline packages/expo-dev-menu/android/src/main/java/expo/modules/devmenu/DevMenuPackage.kt:16]: Same here, probably it should be `BasePackage`

File: packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/DevLauncherPackage.kt (hunk 1 above)
Comment (lukmccall) [reply packages/expo-dev-launcher/android/src/debug/java/expo/modules/devlauncher/DevLauncherPackage.kt:12]: It shouldn't be a BasePackage according to this comment: https://github.com/expo/expo/blob/87e36922e5d3c2ae7422a38451dbd7f6379d8aaa/packages/expo-modules-core/android/src/main/java/expo/modules/core/BasePackage.java#L16

And it has to be a `ReactPackage` too for now. This is only the first step of the migration to the `expo-modules`. Right now, I don't want to change native code much. This pull request only changes how the package is linked.

**Review by esamelson — APPROVED** [review]
👌

**Review by tsapeta — APPROVED** [review]
Looks good, but please fix the CI if it's affected by this PR 😉

**Comment by esamelson** [comment]
Hmm, I've just noticed that the change to tools/src/Packages.ts has caused the unit tests to stop running on CI. What was the reason that change was needed here?

**Pull request merged**